src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs

```diff
@@ -75,7 +75,9 @@ protected override async Task OnFileFound(ProcessRequest processRequest, IDictio
                 }
                 else
                 {
+#pragma warning disable CA1308
```

Comment:
no need to suppress here,  instead  we can use .toUpperInvariant` and change `.mod` -> `.MOD` and `.sum` -> `.SUM` in the switch statement below.